airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/build_image/steps/common.py

```diff
@@ -70,27 +70,28 @@ def image_name(self) -> Tuple:
         return f"airbyte/{self.context.connector.technical_name}"
 
     async def _run(self) -> StepResult:
-        container_variants = list(self.containers.values())
-        _, exported_tar_path = await export_containers_to_tarball(self.context, container_variants)
-        if not exported_tar_path:
-            return StepResult(
-                self,
-                StepStatus.FAILURE,
-                stderr=f"Failed to export the connector image {self.image_name}:{self.image_tag} to a tarball.",
-            )
-        try:
-            client = docker.from_env()
-            response = client.api.import_image_from_file(str(exported_tar_path), repository=self.image_name, tag=self.image_tag)
-            try:
-                image_sha = json.loads(response)["status"]
-            except (json.JSONDecodeError, KeyError):
+        loaded_images = []
+        multi_platforms = len(self.containers) > 1
+        for platform, container in self.containers.items():
+            _, exported_tar_path = await export_container_to_tarball(self.context, container, platform)
+            if not exported_tar_path:
                 return StepResult(
                     self,
                     StepStatus.FAILURE,
-                    stderr=f"Failed to import the connector image {self.image_name}:{self.image_tag} to your Docker host: {response}",
+                    stderr=f"Failed to export the connector image {self.image_name}:{self.image_tag} to a tarball.",
                 )
-            return StepResult(
-                self, StepStatus.SUCCESS, stdout=f"Loaded image {self.image_name}:{self.image_tag} to your Docker host ({image_sha})."
-            )
-        except docker.errors.DockerException as e:
-            return StepResult(self, StepStatus.FAILURE, stderr=f"Something went wrong while interacting with the local docker client: {e}")
+            try:
+                client = docker.from_env()
+                image_tag = f"{self.image_tag}-{platform.replace('/', '-')}" if multi_platforms else self.image_tag
```

Comment:
💅 Can be useful to have title transforms named

```suggestion
                image_tag = _generate_dev_tag(self.image_tag, platform, multi_platforms=multi_platforms)
```

That way you can contain the why

```py
def _generate_dev_tag(image_tag, platform, multi_platforms):
    """
    Explain that multi archecture dev builds are not possible
    """
    return f"{self.image_tag}-{platform.replace('/', '-')}" if multi_platforms else self.image_tag